src/handlers/gestures/gestureBuilder.ts

```diff
@@ -0,0 +1,113 @@
+import { SimpleGesture } from './simpleGestures';
+
+enum Relation {
+  Simultaneous,
+  Exclusive,
+  RequireToFail,
+}
+
+type PendingGesture = {
+  relation: Relation;
+  gesture: SimpleGesture;
+};
+
+export class GestureBuilder {
+  private pendingGestures: PendingGesture[] = [];
+
+  constructor(base: SimpleGesture) {
+    this.addGesture({ relation: Relation.Exclusive, gesture: base });
+  }
+
+  private addGesture(gesture: PendingGesture): GestureBuilder {
+    this.pendingGestures.push(gesture);
+    return this;
+  }
+
+  simultaneousWith(gesture: SimpleGesture): GestureBuilder {
+    return this.addGesture({
+      relation: Relation.Simultaneous,
+      gesture: gesture,
+    });
+  }
+
+  exclusiveWith(gesture: SimpleGesture): GestureBuilder {
+    return this.addGesture({
+      relation: Relation.Exclusive,
+      gesture: gesture,
+    });
+  }
+
+  requireToFail(gesture: SimpleGesture): GestureBuilder {
+    this.pendingGestures.push({
+      relation: Relation.RequireToFail,
+      gesture: gesture,
+    });
+
+    return this;
+  }
+
+  build(): BuiltGesture {
+    const result = new BuiltGesture(this.prepare);
+    result.gestures = this.pendingGestures.map((pending) => pending.gesture);
+    return result;
+  }
+
+  prepare = () => {
+    const simultaneous: number[] = [];
+    const waitFor: number[] = [];
```

Comment:
Isn't that `enum Relation` instead of `number`?

Reply:
No, those arrays store `handlerTags` of gestures with specified relation.

Reply:
I see, so we can consider adding `Tags` suffix to them.

Reply:
Renamed in 8c9bd4380dfe25f2174e83d174d816bbf49688a5